chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -120,7 +120,7 @@
   group "root"
   action :create
   variables(
-    needs_block_migrate: nova[:nova][:use_shared_instance_storage] ? false : true
+    needs_block_migrate: nova.nil? || (nova[:nova][:use_shared_instance_storage] ? false : true)
```

Comment:
`nova[:nova][:use_shared_instance_storage] ? false : true` - I'm not sure why @rhafer put it there, maybe there was some reason I do not see. That's why I prefer not to do it.

As for lazy resources - that seems right, but is it supported in our (quite old) version of chef? I do not see them used anywhere in our recipes.